src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixLegacyResourceDesignerStep.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+using Java.Interop.Tools.Cecil;
+
+using Mono.Linker;
+using Mono.Linker.Steps;
+
+using Mono.Tuner;
+#if ILLINK
+using Microsoft.Android.Sdk.ILLink;
+#endif  // ILLINK
+
+namespace MonoDroid.Tuner
+{
+	public class FixLegacyResourceDesignerStep : LinkDesignerBase
+	{
+		const string DesignerAssemblyName = "_Microsoft.Android.Resource.Designer";
+#if ILLINK
+		protected override void Process ()
+		{
+			cache = Context;
+		}
+#else
+		public FixLegacyResourceDesignerStep (IMetadataResolver cache)
+		{
+			this.cache = cache;
+		}
+
+		readonly
+#endif
+		IMetadataResolver cache;
+		AssemblyDefinition designerAssembly = null;
+		TypeDefinition designerType = null;
+		Dictionary<string, MethodDefinition> lookup;
+
+		protected override void EndProcess ()
+		{
+			if (designerAssembly != null) {
+				LogMessage ($"  Setting Action on {designerAssembly.Name} to Link.");
+				Annotations.SetAction (designerAssembly, AssemblyAction.Link);
+			}
+		}
+
+		protected override void LoadDesigner ()
+		{
+			if (designerAssembly != null)
+				return;
+			var designerNameAssembly = AssemblyNameReference.Parse ($"{DesignerAssemblyName}, Version=1.0.0.0");
+			try {
+				designerAssembly = Resolve (designerNameAssembly);
+			} catch (Mono.Cecil.AssemblyResolutionException) {
+			} catch (System.IO.FileNotFoundException) {
+			}
```

Comment:
Do these exceptions have extra info we could log? or is the null check below enough?

Reply:
I'll add a log for each so we can tell which of those exceptions was raised.